index.js

```diff
@@ -2463,7 +2463,28 @@ export default function(value, nominatim_object, optional_conf_parm) {
                      * use it and ignore lesser specific ones (for the
                      * country)
                      */
-                    return holiday_definitions[location_cc][location_state][type_of_holidays];
+                    var holidays_by_name = {};
+                    var country_holidays = holiday_definitions[location_cc][type_of_holidays];
+                    var state_holidays = holiday_definitions[location_cc][location_state][type_of_holidays];
+                    if (type_of_holidays === 'PH') {
+                        return holiday_definitions[location_cc][location_state][type_of_holidays];
+                        Object.keys(country_holidays || {}).forEach(function(k) {
+                            holidays_by_name[k] = country_holidays[k];
+                        });
+                        Object.keys(state_holidays).forEach(function(k) {
+                            holidays_by_name[k] = state_holidays[k];
+                        });
+                    } else {
+                        (country_holidays || []).forEach(function(holiday) {
+                            holidays_by_name[holiday.name] = holiday;
```

Comment:
Using a dict to store SH definitions won’t really work because the point I use an array for them was that they need to be in order.

Reply:
I finally found time to look into this PR again. It's a bit tricky, also to setup a suitable test environment.

I implemented the chronological merging in 221919a2faee49fdea97ab40453e70d175a398ca. I'm interested in your feedback. 😄